### PR TITLE
Eliminate a few Lens type annotations

### PR DESCRIPTION
**Pull request by jspahrsummers**
One of my [initial reactions](https://twitter.com/jspahrsummers/status/565580761508888576) to Pistachio (besides, “fuck yeah lenses”) was a bit of disappointment at the number of type annotations required, since I hate to duplicate anything for the compiler.

However, I didn't realize that the level of annotation [given in the README](https://github.com/felixjendrusch/Pistachio/blob/1ef81d27df29bb6f9bc41c0f800e83ef28b8481b/README.md#usage) is actually overkill, and that less will still compile perfectly fine.

So, in short, this is an opinionated change intended to counteract the sort of reaction that I had, by reducing code/type duplication in the README and some of Pistacho's tests. Feel free to accept or reject it! :smile:

### CODE REVIEW

**Comment by felixvisee** [comment]
My intention was to provide some additional documentation in the README by explicitly specifying more types than necessary. However, I do see your point and will happily accept your contributions, especially because I consider the resulting code to be of better quality. What do you think about the other type annotations?

**Comment by felixvisee** [comment]
btw, kind of related, you may like the part about improving maintainability of lenses for immutable models by defining a `copy` function, similar to [Scala's copy method on case classes](http://stackoverflow.com/a/7249439), in this short [post](https://github.com/felixjendrusch/blog/blob/master/_posts/2015-02-14-working-with-immutable-models-in-swift.md).

**Comment by jspahrsummers** [comment]
> What do you think about the other type annotations?

Some of the others couldn't be inferred reliably. :confused: 

I didn't run through everything in the README, but I avoided anything that I didn't _know_ could be inferred by the compiler.

**Comment by felixvisee** [comment]
Ok, thanks!
